src/lava/proc/lif/models.py

```diff
@@ -320,3 +320,110 @@ def reset_voltage(self, spike_vector: np.ndarray):
         """Reset voltage of all spiking neurons to 0.
         """
         self.v[spike_vector != 0] = 0  # Reset voltage to 0 wherever we spiked
+
+
+@implements(proc=LIFReset, protocol=LoihiProtocol)
+@requires(CPU)
+@tag('floating_pt')
+class PyLifResetModelFloat(AbstractPyLifModelFloat):
+    """Implementation of Leaky-Integrate-and-Fire neural process with reset
+    in floating point precision.
+    """
+    s_out: PyOutPort = LavaPyType(PyOutPort.VEC_DENSE, float)
+    vth: float = LavaPyType(float, float)
+
+    def __init__(self, proc_params):
+        super(PyLifResetModelFloat, self).__init__(proc_params)
+        self.reset_interval = proc_params['reset_interval']
+        self.reset_offset = (proc_params['reset_offset']) % self.reset_interval
+
+    def spiking_activation(self):
+        """Spiking activation function for LIF.
+        """
+        return self.v > self.vth
+
+    def run_spk(self):
+        """The run function that performs the actual computation during
+        execution orchestrated by a PyLoihiProcessModel using the
+        LoihiProtocol.
+        """
+        # Receive synaptic input
+        a_in_data = self.a_in.recv()
+
+        if (self.time_step % self.reset_interval) == self.reset_offset:
+            self.u *= 0
```

Comment:
Why *= instead of =?

Reply:
Setting it to 0 would make it a scalar with no shape. But what we want is to make it `np.zeros` of same shape.